File: leetcode/go/938.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func rangeSumBST(root *TreeNode, low int, high int) int {
	if root == nil {
		return 0
	}

	if root.Val < low {
		return rangeSumBST(root.Right, low, high)
	}

	if root.Val > high {
		return rangeSumBST(root.Left, low, high)
	}

	return root.Val + rangeSumBST(root.Left, low, high) + rangeSumBST(root.Right, low, high)
}

File: leetcode/python/938.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    count = 0

    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        self.TraverseAndCount(root, low, high)

        return self.count

    def TraverseAndCount(self, node: TreeNode, low: int, high: int):
        if node:
            if node.val >= low and node.val <= high:
                self.count += node.val

            self.TraverseAndCount(node.left, low, high)
            self.TraverseAndCount(node.right, low, high)
